### PR TITLE
Fix testChunkIdxWithDeletedVec

### PR DESCRIPTION
**Pull request by michalkurka**
The original test cannot run reliably in multi-node

### CODE REVIEW

File: h2o-core/src/test/java/water/fvec/VecTest.java [hunk 1]
```diff
@@ -197,26 +197,16 @@ public void map(Chunk c) {
...
-        }
-        @Override
-        public void map(Chunk c) {
-          if (c.vec().get_type() != Vec.T_NUM)
-            throw new IllegalStateException("Expected a numeric Vec");
-        }
-      }.doAll(v);
+      k = v._key;
+      v.remove();
+      v.chunkForChunkIdx(0);
```
Comment (seb-h2o) [inline h2o-core/src/test/java/water/fvec/VecTest.java]: if you expect an exception, you should probably add a `fail()` statement before `catch`.

File: h2o-core/src/test/java/water/fvec/VecTest.java [hunk 2]
```diff
@@ -197,26 +197,16 @@ public void map(Chunk c) {
...
     } catch (Exception e) {
       String msg = e.getMessage();
-      if (msg == null || !msg.contains("Missing chunk") || !msg.contains("is not in DKV"))
```
Comment (seb-h2o) [inline h2o-core/src/test/java/water/fvec/VecTest.java]: maybe checking `e.toString()` instead of `e.getMessage()` would work here?

Comment (michalkurka) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: Can you please explain what issue it addresses?

Comment (seb-h2o) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: I thought that as DistributedException may have been wrapping the original exception without message using basic constructor `public DistributedException(Throwable cause)`, then your test would always fail.

Comment (seb-h2o) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: for example in `RPC` and `MRTask`:
```
        throw (re instanceof DistributedException)?new DistributedException(re.getMessage(),re.getCause()):new DistributedException(re);
```
for some (unknown) reason, the message is propagated only if the cause exception is already a `DistributedException`

Comment (seb-h2o) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: my mistake, didn't see the full logic: by default it's prefixing the cause message...

Comment (michalkurka) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: Okay, I see - that would be for the original code.

Current code won't involve distributed execution - I will always get exactly the exception I am throwing.

The issue with the previous one was not wrapping into DistributedException, it was an NPE in the initialization of the MRTask on the remote nodes.

Comment (seb-h2o) [reply h2o-core/src/test/java/water/fvec/VecTest.java]: yes, I'm using split view (there, it is clear it was a comment on original test).
with unified view, my comment may have been confusing

**Comment by seb-h2o** [comment]
@michalkurka is it ready for merge?

**Pull request merged**